Control/Monad/Accum.hs

```diff
@@ -0,0 +1,247 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | Module: Control.Monad.Accum
+-- Copyright: (C) Koz Ross 2022
+-- License: BSD-3-Clause (see the LICENSE file)
+-- Maintainer: koz.ross@retro-freedom.nz
+-- Stability: Experimental
+-- Portability: GHC only
+--
+-- [Computation type:] Accumulation (either append-only state, or writer with
+-- the ability to read all previous input).
+--
+-- [Binding strategy:] Binding a function to a monadic value monoidally
+-- accumulates the subcomputations (that is, using '<>').
+--
+-- [Useful for:] Logging, patch-style tracking.
+--
+-- [Zero and plus:] None.
+--
+-- [Example type:] @'Accum' w a@
```

Comment:
I actually kind of like this doc style. 👍

Reply:
Can't claim any credit here. I'm just using what already existed for (at least) `MonadCont`.